src/main/java/seedu/duke/Storage.java

```diff
@@ -0,0 +1,60 @@
+package seedu.duke;
+
+import seedu.task.Deadline;
+import seedu.task.Event;
+import seedu.task.Task;
+import seedu.task.TaskList;
+import seedu.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * The storage class deals with loading tasks from the file and saving tasks in the file
+ */
+public class Storage {
+    private final File file;
+
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    public ArrayList<Task> load() throws FileNotFoundException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner s = new Scanner(this.file); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            String str = s.nextLine();
+            char type = str.charAt(4);
+            boolean isDone = str.charAt(1) == 'x';
+            String task = str.substring(7);
+            switch (type) {
+                case 'T':
+                    tasks.add(new Todo(task, isDone));
+                    break;
+                case 'D':
+                    tasks.add(new Deadline(task, isDone));
+                    break;
+                case 'E':
+                    tasks.add(new Event(task, isDone));
+                    break;
```

Comment:
According to the course's coding conventions, çase'statements should not be indented